lib/events.js

```diff
@@ -120,10 +96,61 @@ EventEmitter.prototype.emit = function emit(type) {
       listeners[i].apply(this, args);
   }
 
+  return true;
+}
+
+function $emitError(type, er) {
+  // If there is no 'error' event listener then throw.
+  if (this.domain) {
+    if (!er)
+      er = new Error('Uncaught, unspecified "error" event.');
+    er.domainEmitter = this;
+    er.domain = this.domain;
+    er.domainThrown = false;
+    this.domain.emit('error', er);
+  } else if (er instanceof Error) {
+    throw er; // Unhandled 'error' event
+  } else {
+    throw Error('Uncaught, unspecified "error" event.');
```

Comment:
Think the specification states that will always return a new `Error` object regardless. But since I'm there might as well switch it up to use `new`.